libs/agno/agno/memory/manager.py

```diff
@@ -707,6 +712,204 @@ def _get_first_n_memories(self, user_id: str, limit: Optional[int] = None) -> Li
 
         return sorted_memories_list
 
+    def optimize_memories(
+        self,
+        user_id: Optional[str] = None,
+        token_limit: int = 2000,
+        strategy: Union[str, MemoryOptimizationStrategyType, MemoryOptimizationStrategy] = "summarize",
```

Comment:
We should only support MemoryOptimizationStrategyType and MemoryOptimizationStrategy